src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -992,16 +992,16 @@ public int GatherFields(DvText lineSpan, string path = null, long line = 0)
                                 }
                                 var spanT = Fields.Spans[Fields.Count - 1];
 
-                                // Note that Convert produces NA if the text is unparsable.
-                                DvInt4 csrc = default(DvInt4);
+                                // Note that Convert throws exception the text is unparsable.
+                                int csrc = default;
                                 Conversion.Conversions.Instance.Convert(ref spanT, ref csrc);
```

Comment:
Throwing for this item specifically I think is not what we want to do here, we want to continue to log the bad format for the sparse values if we continue to do so later, right?

Previously for this error there was one failure mode, that resulted in writing to the log.

Right now with this change there are two failure modes... one writes to the log if the index is in some format, the other throws an exception. I'm not sure which is better or worse, but at least it ought to be consistent.

Reply:
I'll catch the exception and keep it consistent with the old behavior. 

---
In reply to: [212045610](https://github.com/dotnet/machinelearning/pull/683#discussion_r212045610) [](ancestors = 212045610)